source/extensions/filters/common/rbac/matchers.cc

```diff
@@ -142,6 +145,26 @@ bool IPMatcher::matches(const Network::Connection& connection, const Envoy::Http
   case DownstreamLocal:
     ip = info.downstreamAddressProvider().localAddress();
     break;
+  case Upstream: {
+    if (!info.filterState().hasDataWithName(StreamInfo::KEY_DYNAMIC_PROXY_UPSTREAM_ADDR)) {
+      ENVOY_LOG_MISC(info, "Did not find dynamic forward proxy cookie. Do you have dynamic "
```

Comment:
Should this be warn or error?
Also I don't think that cookie is the right word.  metadata?  stream info?
And finally maybe make it clear it's not about just having DFP but DFP ahead of rbac

Reply:
👍